src/tools/schema2kotlin.ts

```diff
@@ -126,6 +129,28 @@ abstract class Abstract${particleName} : ${this.opts.wasm ? 'WasmParticleImpl' :
   private getType(type: string): string {
     return this.opts.wasm ? `Wasm${type}` : type;
   }
+
+  private getQueryType(connection: HandleConnectionSpec): string | null {
+    if (!connection.type.isCollectionType()) {
+      return 'Unit';
+    }
+    const type = connection.type.collectionType;
+    if (!(type instanceof EntityType)) {
+      return 'Unit';
+    }
+    const refinement = type.entitySchema.refinement;
+    if (!refinement) {
+      return null;
+    }
+    const queryArgType: Primitive = refinement.getQueryNames().get('?');
+    switch (queryArgType) {
+      case 'Text': return 'String';
+      case 'Number': return 'Int';
```

Comment:
Kotlin Arcs uses all numbers as `Double`s, instead of `Int`s

Reply:
Thanks good catch.